layers/+lang/extempore/README.org

```diff
@@ -0,0 +1,61 @@
+#+TITLE: Extempore layer
+#+TAGS: layer|lisp|programming|music
+
+# The maximum height of the logo should be 200 pixels.
+# [[img/extempore.png]]
+
+# TOC links should be GitHub style anchors.
+* Table of Contents                                        :TOC_4_gh:noexport:
+- [[#description][Description]]
+  - [[#features][Features:]]
+- [[#install][Install]]
+  - [[#configuration][Configuration]]
+- [[#key-bindings][Key bindings]]
+
+* Description
+This layer adds support for the [[https://github.com/digego/extempore][Extempore]] programming environment.
+
+** Features:
+  - run Extempore (inferior extempore buffer)
+  - connect to & evaluate code
+  - eldoc support
+
+* Install
+To use this configuration layer, add it to your =~/.spacemacs=. You will need to
+add =extempore= to the existing =dotspacemacs-configuration-layers= list in this
+file.
+
+** Configuration
+
+The only configuration variable you might want to set out-of-the-box is
+=extempore-path=, so that you don't have to answer the "which directory?" prompt
+every time you call =switch-to-extempore=. You can set it when you add
+=extempore= to your =dotspacemacs-configuration-layers= like so:
+
+#+BEGIN_SRC
+(extempore :variables extempore-path "/path/to/extempore/")
+#+END_SRC
+
+Alternately, you could set it elsewhere in your init file, or through the
+=customize= interface.
+
+There are a few other config variables to tweak, but in most cases the defaults
+will be ok if you're just starting out. If you want to see them all, open up
+=M-x customize= and look in the =extempore= group.
+
+* Key bindings
+
+| Key Binding | Description             |
+|-------------+-------------------------|
+| ~, c c~     | Switch to Extempore     |
+| ~, c j~     | Connect to Extempore    |
+| ~, e f~     | Eval current definition |
+| ~, e e~     | Eval last sexp          |
+| ~, e r~     | Eval region             |
+| ~, e b~     | Eval buffer             |
+
+# Use GitHub URLs if you wish to link a Spacemacs documentation file or its heading.
```

Comment:
Remove layer comments, these are just for guidance during the creation process and should not stay in real layer code.